sql/2019/02_CSS/02_06.sql

```diff
@@ -0,0 +1,86 @@
+#standardSQL
+# 02_06: % of sites that use each color format
+CREATE TEMPORARY FUNCTION getAllValues(css STRING)
+RETURNS STRUCT<hsl BOOLEAN, hsla BOOLEAN, rgb BOOLEAN, rgba BOOLEAN, hex BOOLEAN> LANGUAGE js AS '''
+try {
+  var getColorFormat = (value) => {
+    value = value.toLowerCase();
+    if (value.includes('hsl(')) {
+      return 'hsl';
+    }
+    if (value.includes('hsla(')) {
+      return 'hsla';
+    }
+    if (value.includes('rgb(')) {
+      return 'rgb';
+    }
+    if (value.includes('rgba(')) {
+      return 'rgba';
+    }
+    if (value.match(/#\\d{3,}/)) {
+      return 'hex';
+    }
+    return null;
```

Comment:
Oh, missed this comment. I'd personally find that interesting, but it's not one of the formats listed in the metric description. It'd also be a bit trickier since the names don't follow a well-defined pattern; we'd need to maintain a safelist of known color values.